interactors/nonceHandlerV2/addressNonceHandler.go

```diff
@@ -193,7 +190,8 @@ func (anh *addressNonceHandler) isTxAlreadySent(tx *transaction.FrontendTransact
 	anh.mut.RLock()
 	defer anh.mut.RUnlock()
 	for _, oldTx := range anh.transactions {
-		isTheSameReceiverDataValue := oldTx.Receiver == tx.Receiver &&
+		isTheSameReceiverDataValue := tx.Nonce == oldTx.Nonce &&
```

Comment:
this still doesn't cover all the situations. maybe I want to send the transaction again, but by using usernames, or whatever field's alteration. safest way would be to compare the hashes of the transactions. 
also, the var name should be `areTransactionsTheSame` since we now compare the nonce as well
also, oldTx.Nonce == tx.Nonce, for consistency

Reply:
will change but will only test the nonce. In this case the component can be used to resend competing transactions only if the gas is higher on the newest transaction, regardless what else changed.

Reply:
```suggestion
		isTheSameNonceReceiverDataValue := tx.Nonce == oldTx.Nonce &&
```

Reply:
Agree here, I think we should use hashes

Reply:
Does not work in the case we resend the same transaction but with a smaller value for gas price as the last one. It will circumvent the check for the gas price. Will test only for the nonce as it suffice

Reply:
We could copy both txs without gas prices and compare hashes. Similar the way we verify signatures on headers